File: ShadowFrameworkLite20/src/shadow/image/bitmaps/data/SFFunction2DBitmapData.java
package shadow.image.bitmaps.data;

import shadow.image.SFBitmap;
import shadow.image.bitmaps.SFBitmapFunction;
import shadow.image.bitmaps.SFFunction2DBitmap;
import shadow.renderer.data.SFDataAsset;
import shadow.renderer.data.SFLibraryReference;
import shadow.system.data.SFDataCenterListener;
import shadow.system.data.SFNamedParametersObject;

public class SFFunction2DBitmapData extends SFBitmapData{

	private SFLibraryReference<SFBitmapFunction> function=new SFLibraryReference<SFBitmapFunction>(); 
	
	public SFFunction2DBitmapData() {
		super();
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("width", width);
		parameters.addObject("height", height);
		parameters.addObject("function", function);
		setData(parameters);
	}
	
	public void setFunction(SFDataAsset<SFBitmapFunction> function) {
		this.function.setDataset(function);
	}

	@Override
	protected SFBitmap buildResource() {
		final SFFunction2DBitmap bitmap=new SFFunction2DBitmap(width.getShortValue(), height.getShortValue(), false, null);
		function.retrieveDataset(new SFDataCenterListener<SFDataAsset<SFBitmapFunction>>() {
			@Override
			public void onDatasetAvailable(String name, SFDataAsset<SFBitmapFunction> dataset) {
				bitmap.setFunction(dataset.getResource());
			}
		});
		return bitmap;
	}
}

File: ShadowFrameworkJS/src/shadow/image/bitmaps/data/SFFunction2DBitmapData.js


function SFFunction2DBitmapData(){
	
	this.func=new SFLibraryReference(); 
	
	this.width=new SFShort();
	this.height=new SFShort();
	
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.width);
	parameters.addObject(this.height);
	parameters.addObject(this.func);
	this.setData(parameters);
}

inherit(SFFunction2DBitmapData,SFBitmapData);


SFFunction2DBitmapData.prototype["buildResource"]=function(){
	var bitmap=new SFFunction2DBitmap(this.width.getShortValue(), this.height.getShortValue(), false, null);
	var dataset=this.func.retrieveDataset();//new SFDataCenterListener<SFDataAsset<SFBitmapFunction>>() {
	bitmap.setFunction(dataset.getResource());
	return bitmap;
};


SFFunction2DBitmapData.prototype["generateNewDatasetInstance"]=function(){
	return new SFFunction2DBitmapData();
};
